dask_sql/physical/rel/logical/cross_join.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import time
+from typing import TYPE_CHECKING
+
+import dask.dataframe as dd
+
+from dask_sql.datacontainer import ColumnContainer, DataContainer
+from dask_sql.physical.rel.base import BaseRelPlugin
+
+if TYPE_CHECKING:
+    import dask_sql
+    from dask_planner.rust import LogicalPlan
+
+logger = logging.getLogger(__name__)
+
+
+class DaskCrossJoinPlugin(BaseRelPlugin):
+    """
+    While similar to `DaskJoinPlugin` a `CrossJoin` has enough of a differing
+    structure to justify its own plugin. This in turn limits the number of
+    Dask tasks that are generated for `CrossJoin`'s when compared to a
+    standard `Join`
+    """
+
+    class_name = "CrossJoin"
+
+    def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContainer:
+        # We now have two inputs (from left and right), so we fetch them both
+        dc_lhs, dc_rhs = self.assert_inputs(rel, 2, context)
+
+        df_lhs = dc_lhs.df
+        df_rhs = dc_rhs.df
+
+        # Create a 'key' column in both DataFrames to join on
+        cross_join_key = "cross_join_key_" + str(time.time_ns())
+        df_lhs[cross_join_key] = 1
+        df_rhs[cross_join_key] = 1
+
+        result = dd.merge(df_lhs, df_rhs, on=cross_join_key).drop(cross_join_key, 1)
```

Comment:
I came across changes in #150 which seemed to change the implementation of cross joins (in join.py), thought I don't completely understand why the updated version has better perf. It might be worth opening an issue to test cross join perf and memory usage and see if the approach in #150 improves it.

Reply:
Yeah I've wondered why we opted to construct the cross join graph manually rather than depend on `dd.merge` for a while now - would be nice to get some insight on the impact it has performance-wise